etcdserver/storage.go

```diff
@@ -109,14 +108,6 @@ func readWAL(waldir string, snap walpb.Snapshot) (w *wal.WAL, id, cid types.ID,
 // It must ensure that, after upgrading, the most recent version is present.
 func upgradeDataDir(baseDataDir string, name string, ver version.DataDirVersion) error {
 	switch ver {
-	case version.DataDir0_4:
-		plog.Infof("converting v0.4 log to v2.0")
-		err := migrate.Migrate4To2(baseDataDir, name)
-		if err != nil {
-			plog.Fatalf("failed to migrate data-dir (%v)", err)
-			return err
-		}
-		fallthrough
```

Comment:
We will do a lot of work to warn user about not upgrading from 0.4 to 2.2, but we cannot stop every user to do it. If one user does migrate from 0.4 to 2.2, he could be totally confused, which makes troubles.

etcd v0.4 -> etcd v2.2 doesn't have version enforcement, so we cannot stop upgrading multiple versions.

I want users to get explicit error if possible to save the time of both sides.